Prefer ansible.builtin.include_role to a roles block

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Modifies the Ansible code to prefer `ansible.builtin.include_role` to a `roles` block in playbooks and playbook fragments
- Alphabetizes the attributes for each entry in `src/requirements.txt`

## 💭 Motivation and context ##

- `ansible.builtin.include_role` is more flexible, as described in #198.
- We prefer to alphabetize YAML attributes unless there is a compelling reason not to.

Resolves #198.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

jsf9k committed:
Prefer ansible.builtin.include_role to a roles block

The former is more flexible, as described in cisagov/skeleton-packer#198.

jsf9k committed:
Alphabetize attributes

dav3r reviewed (approved):
👍 👍

mcdonnnj commented on src/example.yml:
```diff
@@ -3,5 +3,7 @@
-  roles:
-    - example
+  tasks:
+    - name: Install cisagov/example
```
I would like a better description here because that Ansible role [does nothing](https://github.com/cisagov/skeleton-ansible-role/blob/develop/tasks/main.yml) currently.

jsf9k replied:
Please see commit ff2984a.

mcdonnnj commented on src/python.yml:
```diff
@@ -3,7 +3,13 @@
-    - python
-    - remove_python2
+  tasks:
+    - name: Install pip3
+      ansible.builtin.include_role:
+        name: pip
+    - name: Install Python 3
+      ansible.builtin.include_role:
+        name: python
```
Thoughts on flipping these now that they're no longer a list?

jsf9k replied:
Please see commit 2c1b7a1.

dav3r replied:
Can you please add a comment here similar to (or the same as) the commit message you wrote in https://github.com/cisagov/skeleton-packer/commit/2c1b7a1269f28c6488380550e87c6658b7ea6a5e?  That will be a helpful reminder to us and others why the order matters here.

jsf9k replied:
@dav3r, please have a look at commit 45ea09c.

dav3r replied:
Looks great, thanks!

mcdonnnj reviewed (commented):
I like the direction this PR is taking us but I have a request and a question for your consideration.

jsf9k committed:
Fix misleading task name

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install Python before pip

If pip is installed first, then the OS _could_ pull different Python
packages than what would be installed via the
cisagov/ansible-role-python role; hence, the new ordering is more
controlled.

Co-authored-by: Nicholas McDonnell <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add a comment explaining why Python is installed before pip

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>
